userspace/engine/rule_reader.cpp

```diff
@@ -0,0 +1,325 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "rule_reader.h"
+
+#define THROW(cond, err)    { if (cond) { throw falco_exception(err); } }
+
+static rule_loader::mark yaml_get_mark(
```

Comment:
Don't feel strongly about this, but why the static functions as compared to private methods?

Reply:
This was simply a matter of not over-complicating the class definition by adding helper methods that are just used internally. If you are strong on this point, I can turn them into private methods.

Reply:
No, it's okay. That's more of a style thing and I can work with either, as long as the static functions don't end up passing too many args that should be instance variables. That's not the case here.